src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java

```diff
@@ -162,10 +187,27 @@ SELECT COUNT(r.id)
     @EntityGraph(type = LOAD, attributePaths = { "feedbacks" })
     List<Result> findAllWithEagerFeedbackByAssessorIsNotNullAndParticipation_ExerciseIdAndCompletionDateIsNotNull(Long exerciseId);
 
-    @Query("SELECT COUNT(DISTINCT p) FROM Participation p left join p.results r WHERE p.exercise.id = :exerciseId AND r.assessor IS NOT NULL AND r.assessmentType IN :types AND r.rated = TRUE AND r.completionDate IS NOT NULL AND (p.exercise.dueDate IS NULL OR r.submission.submissionDate <= p.exercise.dueDate)")
+    @Query("""
+            SELECT COUNT(DISTINCT p) FROM Participation p left join p.results r
```

Comment:
Why a left join here? I think you should filter out all participations with results already at the join step.

Reply:
Here too.